xds/internal/balancer/edsbalancer/eds_impl.go

```diff
@@ -485,5 +501,18 @@ func (d *dropPicker) Pick(info balancer.PickInfo) (balancer.PickResult, error) {
 	}
 	// TODO: (eds) don't drop unless the inner picker is READY. Similar to
 	// https://github.com/grpc/grpc-go/issues/2622.
-	return d.p.Pick(info)
+	pr, err := d.p.Pick(info)
+	if d.counter != nil && err == nil {
+		if err := d.counter.StartRequest(); err != nil {
```

Comment:
`StartRequest` needs to happen before the Pick, so we return early when the max count is reached.

Reply:
Made this change.